src/rulesql.yrl

```diff
@@ -410,6 +405,20 @@ parsetree_with_tokens(Sql0) ->
                 {error, Error} -> {parse_error, {Error, Toks}}
             end;
         {error, Error, _} -> {lex_error, Error}
+    end;
+parsetree_with_tokens({error, Error}) -> {parse_error, Error};
+parsetree_with_tokens(Sql) -> parsetree_with_tokens(sql_clean(Sql)).
```

Comment:
I suggest to add another function before calling the parsetree_with_tokens/1, which first calls sql_clean/1 and then calls parsetree_with_tokens/1 if the Sql is valid.